dashboard/src/components/DeploymentForm/DeploymentFormBody/BasicDeploymentForm/TabularSchemaEditorTable/Params/SliderParam.tsx

```diff
@@ -6,77 +6,106 @@ import { CdsInput } from "@cds/react/input";
 import { CdsRange } from "@cds/react/range";
 import Column from "components/js/Column";
 import Row from "components/js/Row";
-import { toNumber } from "lodash";
+import { isEmpty, toNumber } from "lodash";
 import { useState } from "react";
-import { IBasicFormParam } from "shared/types";
-import { basicFormsDebounceTime } from "shared/utils";
+import { validateValuesSchema } from "shared/schema";
+import { IAjvValidateResult, IBasicFormParam } from "shared/types";
+import { basicFormsDebounceTime, getOptionalMin, getStringValue } from "shared/utils";
 
 export interface ISliderParamProps {
   id: string;
   label: string;
   param: IBasicFormParam;
   unit: string;
-  step: number;
   handleBasicFormParamChange: (
     p: IBasicFormParam,
   ) => (e: React.FormEvent<HTMLInputElement>) => void;
 }
 
 export default function SliderParam(props: ISliderParamProps) {
-  const { handleBasicFormParamChange, id, label, param, step } = props;
+  const { handleBasicFormParamChange, id, label, param } = props;
+
+  const initCurrentValue = () =>
+    toNumber(param.currentValue) ||
+    toNumber(param.exclusiveMinimum) ||
+    toNumber(param.minimum) ||
+    0;
+
+  const [validated, setValidated] = useState<IAjvValidateResult>();
+  const [currentValue, setCurrentValue] = useState(initCurrentValue());
 
-  const [currentValue, setCurrentValue] = useState(
-    toNumber(param.currentValue) || toNumber(param.minimum) || 0,
-  );
   const [isValueModified, setIsValueModified] = useState(false);
   const [timeout, setThisTimeout] = useState({} as NodeJS.Timeout);
 
+  const step =
+    param?.multipleOf || (param.schema?.type === "number" || param?.type === "number" ? 0.5 : 1);
+
   const onChange = (e: React.FormEvent<HTMLInputElement>) => {
     setCurrentValue(toNumber(e.currentTarget.value));
-    setIsValueModified(toNumber(e.currentTarget.value) !== param.currentValue);
+    setIsValueModified(toNumber(e.currentTarget.value) !== toNumber(param.currentValue));
+
+    // twofold validation: using the json schema (with ajv) and the html5 validation
+    setValidated(validateValuesSchema(e.currentTarget.value, param.schema));
+    e.currentTarget?.reportValidity();
```

Comment:
Hopefully these can't oppose each other... :P

Reply:
haha, right. In theory, the html5 validation should be a subset of the schema validation, since we are manually adding stuff like `max`, `min`, etc but we are not considering more complex schemas using `anyOf`, `allOf`, etc. So the browser validation should be able to provide quick feedback in a native way, whereas the ajv validation will prompt the schema validation error, which could be more difficult to understand for non-experienced users.